app/web/src/newhotness/Explore.vue

```diff
@@ -859,16 +730,11 @@ const onU = (e: KeyDetails["u"]) => {
   e.preventDefault();
 
   if (showGrid.value) {
-    if (!interactionTargetComponentId.value) return;
-    const targetComponent = filteredComponents.find(
-      (comp) => comp.id === interactionTargetComponentId.value,
-    );
-    if (
-      targetComponent &&
-      upgrade(targetComponent.schemaId, targetComponent.schemaVariantId).value
-    ) {
+    if (!focusedComponent.value) return;
+
+    if (upgradeableComponents.value.has(focusedComponent.value.id)) {
```

Comment:
The functionality here changed slightly to use the list for performance reasons. We should consider moving this to `ComponentInList` or making a new type in the future.